packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -159,6 +159,7 @@ async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
 
 const PAYLOAD_CACHE = (process.env.NUXT_PAYLOAD_EXTRACTION && process.env.prerender) ? new Map() : null // TODO: Use LRU cache
 const PAYLOAD_URL_RE = /\/_payload(\.[a-zA-Z0-9]+)?.js(\?.*)?$/
+const ROOT_NODE_REGEX = /^<div id="__nuxt">([\s\S]*)<\/div>$/
```

Comment:
Unfortunately, we allow the id + tag to be customised (`app.rootId` and `app.rootTag`) so this regexp will not always work, but we do have access above on:

https://github.com/nuxt/nuxt/blob/859fbd0add7e97633c3350b9baf3aeb271930081/packages/nuxt/src/core/runtime/nitro/renderer.ts#L17

Reply:
thanks !